bokeh/util/token.py

```diff
@@ -147,6 +150,9 @@ def get_token_payload(token: str) -> Any:
         dict
     """
     decoded = json.loads(_base64_decode(token.split('.')[0]))
+    if '__zlib__' in decoded:
+        decompressed = zlib.decompress(_base64_decode(decoded['__zlib__']))
+        decoded.update(json.loads(decompressed))
```

Comment:
shouldn't we `del decoded['__zlib__']` here for good measure? We might also want to document that `__zlib__` is a forbidden key (and also check explicitly for it above during encoding and raise a meaningful error)

Reply:
Yep, that all makes sense. Will try to finish this off this weekend.